react/features/prejoin/components/PrejoinApp.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+
+import conference from '../../../../conference';
+import { BaseApp } from '../../../features/base/app';
+import { setConfig } from '../../base/config';
+import { configureInitialDevices, getAvailableDevices } from '../../base/devices';
+import { DialogContainer } from '../../base/dialog';
+import { initPrejoin } from '../actions';
+
+import Prejoin from './Prejoin';
+
+type Props = {
+
+    /**
+     * Flag signaling if the device status is visible or not.
+     */
+    deviceStatusVisible: boolean,
+
+    /**
+     * If join by phone button should be visible.
+     */
+    hasJoinByPhoneButton: boolean,
+
+    /**
+     * Joins the current meeting.
+     */
+    joinConference: Function,
+
+    /**
+     * Joins the current meeting without audio.
+     */
+    joinConferenceWithoutAudio: Function,
+
+    /**
+     * The name of the user that is about to join.
+     */
+    name: string,
+
+    /**
+     * Updates settings.
+     */
+    updateSettings: Function,
+
+    /**
+     * The name of the meeting that is about to be joined.
+     */
+    roomName: string,
+
+    /**
+     * Sets visibility of the prejoin page for the next sessions.
+     */
+    setSkipPrejoin: Function,
+
+    /**
+     * Sets visibility of the 'JoinByPhoneDialog'.
+     */
+    setJoinByPhoneDialogVisiblity: Function,
+
+    /**
+     * Indicates whether the avatar should be shown when video is off
+     */
+    showAvatar: boolean,
+
+    /**
+     * Flag signaling the visibility of camera preview.
+     */
+    showCameraPreview: boolean,
+
+    /**
+     * Flag signaling the visibility of join label, input and buttons
+     */
+    showJoinActions: boolean,
+
+    /**
+     * If 'JoinByPhoneDialog' is visible or not.
+     */
+    showDialog: boolean,
+
+    /**
+     * Used for translation.
+     */
+    t: Function,
+
+    /**
+     * The JitsiLocalTrack to display.
+     */
+    videoTrack: ?Object,
+};
+
+/**
+ * Wrapper application for prejoin.
+ *
+ * @extends BaseApp
+ */
+export default class PrejoinApp extends BaseApp<Props> {
+    _init: Promise<*>;
+
+    /**
+     * Navigates to {@link Prejoin} upon mount.
+     *
+     * @returns {void}
+     */
+    componentDidMount() {
+        super.componentDidMount();
+
+        this._init.then(async () => {
+            const { dispatch } = this.state.store;
+            const { showAvatar, showJoinActions } = this.props;
+
+            super._navigate({
+                component: Prejoin,
+                props: {
+                    showAvatar,
+                    showJoinActions
+                }
+            });
+
+            dispatch(setConfig({ prejoinPageEnabled: true }));
+
+            await dispatch(configureInitialDevices());
+            await dispatch(getAvailableDevices());
+
+            const { tryCreateLocalTracks, errors } = conference.createInitialLocalTracks(
+                conference._getInitialOptions()
```

Comment:
I'd like us to not use `conference.js` here at all. Even if that means duplicating a bit of code. That file is legacy and the less we add to it or depend on it the better.

Reply:
done